test/clj/rems/application/test_model.clj

```diff
@@ -274,262 +299,348 @@
+                                                                                                 :default "non-localized title"}
+                                                                                 :license/attachment-id {:en 3301
+                                                                                                         :fi 3302
+                                                                                                         :default 3300}
+                                                                                 :license/attachment-filename {:en "en filename"
+                                                                                                               :fi "fi filename"
+                                                                                                               :default "non-localized filename"}
+                                                                                 :license/start (DateTime. 100)
+                                                                                 :license/end nil
+                                                                                 :license/expired false
+                                                                                 :license/enabled true
+                                                                                 :license/archived false}]})]
+                        (is (= expected-application (apply-events events)))
+
+                        (testing "> approved"
+                          (let [events (conj events
+                                             {:event/type :application.event/approved
+                                              :event/time (DateTime. 4000)
+                                              :event/actor "handler"
+                                              :application/id 1
+                                              :application/comment "looks good"})
+                                expected-application (merge expected-application
+                                                            {:application/last-activity (DateTime. 4000)
+                                                             :application/events events
+                                                             :application/state :application.state/approved})]
+                            (is (= expected-application (apply-events events)))
+
+                            (testing "> licenses accepted"
+                              (let [events (conj events
+                                                 {:event/type :application.event/licenses-accepted
+                                                  :event/time (DateTime. 4500)
+                                                  :event/actor "applicant"
+                                                  :application/id 1
+                                                  :application/accepted-licenses #{30 31 32 33}})
```

Comment:
could you have two different users accept different sets of licenses here please?

Reply:
There's also the case that how to interpret accepting licenses multiple times. If a user first accepts licenses 30 and 31, and then later accepts 32 and 33, are now all four licenses accepted or only the last two?

Reply:
I added two users case and also one with overwriting the previous accepted licenses